packages/swing-store/src/bundleStore.js

```diff
@@ -104,17 +117,21 @@ export function makeBundleStore(db, ensureTxn, noteExport = () => {}) {
 
   /**
    * @param {string} bundleID
-   * @returns {EndoZipBase64Bundle}
+   * @returns {Bundle}
    */
   function getBundle(bundleID) {
-    bundleID.startsWith('b1-') || Fail`invalid bundleID ${q(bundleID)}`;
     const rawBundle = sqlGetBundle.get(bundleID);
     rawBundle || Fail`bundle ${q(bundleID)} not found`;
-    return harden({
-      moduleFormat: 'endoZipBase64',
-      endoZipBase64Sha512: bundleID.substring(3),
-      endoZipBase64: encodeBase64(rawBundle),
-    });
+    if (bundleID.startsWith('b0-')) {
+      return harden(JSON.parse(rawBundle));
+    } else if (bundleID.startsWith('b1-')) {
+      return harden({
+        moduleFormat: 'endoZipBase64',
+        endoZipBase64Sha512: bundleID.substring(3),
+        endoZipBase64: encodeBase64(rawBundle),
+      });
+    }
+    throw Fail`unsupported BundleID ${q(bundleID)}`;
```

Comment:
This throws on an unsupported bundle type after having successfully read the bundle from the database.  Something seems off about that.  I think we should be checking the bundleID/type before attempting the read.

Reply:
Yeah, it was that that or copy the `sqlGetBundle.get()` into each arm, or have a `if (!bundleID.startsWith('b0-') || !bundleID.startsWith('b1-')) { throw.. }` at the start of the function and then ask the reader to remember that the check was already done by the time they get to the bottom of the arms. This seemed the most readable.

Reply:
meh, I'll change it to duplicate the `sqlGetBundle`, marginally cleaner that way

Reply:
Ok, right, if this error message ever got printed, it would raise the question "if it's so unsupported, how did that bundle get into the DB in the first place?". 

What if I change the error message to "somehow found unsupported BundleID in DB" ?